### PR TITLE
Feature/restorer

### PR DESCRIPTION
**Pull request by enricgalceran**
Restorer script. Is there any reason why github_restorer and github_archiver should be private? @HannesSommer @hitzg

### CODE REVIEW

**Comment by hitzg** [comment]
I don't think so. The only thing which could be an issue is that the blacklist.txt and whitelist.txt files contain the names of private repos (which whould not be known to the public otherwise). But I'm pretty sure we can consider that a technicality :stuck_out_tongue_winking_eye:

**Comment by enricgalceran** [comment]
OK,  so I'll keep the scripts public, and move the white and blacklists to -private.

File: github/github_restorer.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+        repo_has_wiki = os.path.exists(os.getcwd() + "/" + args.repository + \
+                        ".wiki.git")
+
+        # Restoring. First create repo
+        repo = g.get_organization(args.organization).create_repo(\
+               args.repository, private=True, has_wiki=repo_has_wiki)
+
+        # Push
+        ret = subprocess.call("cd " + args.repository + ".git && git push",
```
Comment (HannesSommer) [inline github/github_restorer.py:117]: So the `--all` is implicit in case of a bare repo?

**Comment by HannesSommer** [comment]
I guess that would be a good moment to add the private as submodule of the public. @enricgalceran  are you going to do that for moving the black/white lists?

File: github/github_restorer.py (hunk 1 above)
Comment (enricgalceran) [inline github/github_restorer.py:117]: @HannesSommer I believe so. At least --all is incompatible with --mirror (we clone repos with --mirror), so passing --all here results in an error.

**Comment by enricgalceran** [comment]
What I've done now is to put dummy white/black lists in the public (check it out). Wouldn't you mean create a submodule in the private repo importing content from the public?

**Comment by HannesSommer** [comment]
As discussed, there are both options. 
What you suggest is nicer for the public-only users and what I suggest can make things easier when developing new tools in the public that require private stuff because then they "know" where to find the private details. Otherwise you always need some extra logic in the private to use the code in the public and point it from outside to the private content. 
But let's just start with your suggestion. I can start that on the git archive server. Where did you put the former content of white/black lists? I couldn't find it. Or shall I take it from the public history?

File: github/github_restorer.py (hunk 1 above)
Comment (HannesSommer) [inline github/github_restorer.py:117]: Ah, interesting as the clone is a mirror clone (the remote is a mirror remote), when pushing `--mirror` is already implicit. Great!

**Comment by enricgalceran** [comment]
Forgot to add them @HannesSommer. They live now inside the 'github_archive_data' directory in the private repo.

**Pull request merged**